### PR TITLE
feat(vps.dashboard): inform VPS customers for planned migration

### PR DESCRIPTION
**Pull request by mohammed-zahaf**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-7428
| License          | BSD 3-Clause

## Description

Display banner based on the right zone `/.../vps/migrationStein`
### :flags: Translations

- [x] TRANS-37847

### CODE REVIEW

**Review by ganeshkumar1989 — DISMISSED** [review]

**Review by Jisay — APPROVED** [review]

File: packages/manager/modules/vps/src/dashboard/vps-dashboard.component.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default {
     vpsMigrationTask: '<',
     vpsMigrationTaskInError: '<',
     vpsUpgradeTask: '<',
-    canDisplayVpsAnnouncementBanner: '<',
+    isVpsMaintenance: '<',
```
Comment (antleblanc) [inline packages/manager/modules/vps/src/dashboard/vps-dashboard.component.js:45]: IMHO:

```suggestion
    isInMaintenance: '<',
```

sounds more correct. Isn't it?

Comment (mohammed-zahaf) [reply packages/manager/modules/vps/src/dashboard/vps-dashboard.component.js:45]: i think first one is more explicit.
thanks for suggestion.

**Review by radireddy — DISMISSED** [review]

**Review by JeremyDec — APPROVED** [review]

File: packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js [hunk 2]
```diff
@@ -189,13 +189,8 @@ export default /* @ngInject */ ($stateProvider) => {
       },
       breadcrumb: () => null,
       trackingPrefix: () => 'vps::detail::dashboard',
-      canDisplayVpsAnnouncementBanner: /* @ngInject */ (ovhFeatureFlipping) => {
-        const vpsAnnouncementBannerId = 'vps:vps-announcement-banner';
```
Comment (cbourgois) [inline packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js:193]: `vps:vps-announcement-banner` is not used anymore ?

Comment (mohammed-zahaf) [reply packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js:193]: No, i will reuse it later.
It's a temporary changes time to found a way to deals things properly with API side.

File: packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js [hunk 3]
```diff
@@ -154,14 +154,14 @@ export default /* @ngInject */ ($stateProvider) => {
...
-          .then((newRegionFeature) =>
-            newRegionFeature.isFeatureAvailable(vpsAnnouncementBannerId),
-          );
-      },
+
+      stein: /* @ngInject */ ($http, stateVps) =>
+        $http
+          .get('/vps/migrationStein')
+          .then(({ data }) => data)
+          .then((steins) => steins.find(({ zone }) => zone === stateVps.zone)),
```
Comment (varun257) [inline packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js:162]: Isn't `data` same as `steins` here ?

Comment (mohammed-zahaf) [reply packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js]: Yep, got it.
Changed

**Review by varun257 — APPROVED** [review]

**Review by radireddy — DISMISSED** [review]

**Review by varun257 — APPROVED** [review]

**Comment by mohammed-zahaf** [comment]
Hello @antleblanc, can you please prod this feature, it's bit urgent.
If you need more info let me know, thanks

**Review by antleblanc — CHANGES_REQUESTED** [review]
@mohammed-zahaf Some checks were not successful.

Could you please attempt to rebase on `master` in order to pass all checks?

Thanks!

File: packages/manager/modules/vps/src/dashboard/vps-dashboard.component.js (hunk 1 above)
Comment (antleblanc) [reply packages/manager/modules/vps/src/dashboard/vps-dashboard.component.js:51]: Not sure that's the correct way to express if there's a pending maintenance for the given virtual private server.

cc @marie-j

**Comment by mohammed-zahaf** [comment]
Hello @antleblanc, i rebased on master.
The issue related to "duplicate me" key has been fixed.
So no issue.

Thanks

File: packages/manager/modules/vps/src/dashboard/components/vps-announcement-banner/component.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
-import controller from './controller';
 import template from './template.html';
 
 const component = {
-  controller,
+  bindings: {
+    stein: '<',
```
Comment (marie-j) [inline packages/manager/modules/vps/src/dashboard/components/vps-announcement-banner/component.js:5]: What does `stein` mean ? Is it a boolean ? In that case maybe `isSteinRelease` ?

File: packages/manager/modules/vps/src/dashboard/vps-dashboard.component.js (hunk 1 above)
Comment (marie-j) [reply packages/manager/modules/vps/src/dashboard/vps-dashboard.component.js:51]: isVpsMaintenance would imply that what you're describing is a maintenance and whether it is applied to vps whereas in fact you are describe is a vps which might be in maintenance or not therefore `isInMaintenance` is more accurate

File: packages/manager/modules/vps/src/dashboard/components/vps-announcement-banner/component.js (hunk 4 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/vps/src/dashboard/components/vps-announcement-banner/component.js:5]: it's name used by API.

File: packages/manager/modules/vps/src/dashboard/vps-dashboard.component.js (hunk 1 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/vps/src/dashboard/vps-dashboard.component.js:51]: `isInMaintenance` is not explicit by reading it first times, we have to read more code to understand what is meaning.
Regarding PCI team, the banner is here to inform customer that his VPS is in maintenance and he will probably enconter issues during it.

For me `isVpsMaintenance` is explicit.

**Review by JeremyDec — APPROVED** [review]

**Pull request merged**